### PR TITLE
feat(one-app-bundler): add node 16 support

### PR DESCRIPTION
**Pull request by 10xLaCroixDrinker**
#### _Describe your changes in detail_

Removed the deprecated `fibers` package and updated a peerDep to support node 16.

#### _Why is this change required? What issue does it resolve?_

To make One App compatible with node 16

#### _How does your change affect the rest of the code._ 

Added in one-app and its prod-sample modules and ran all tests.
Existing unit tests

#### Check boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [x] Dependency update
- [ ] Security update

#### Check boxes that apply:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [x] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: packages/holocron-dev-server/package.json [hunk 1]
```diff
@@ -77,7 +77,7 @@
     "redux": "^4.0.5",
     "reselect": "^4.0.0",
     "sass": "^1.26.10",
-    "sass-loader": "^11.0.1",
+    "sass-loader": "^9.0.3",
```
Comment (10xLaCroixDrinker) [inline packages/holocron-dev-server/package.json:80]: This matches what is actually in the bundler

File: packages/one-app-bundler/webpack/loaders/common.js [hunk 2]
```diff
@@ -67,9 +66,6 @@ const sassLoader = () => ({
   loader: 'sass-loader',
   options: {
     implementation: dartSass,
-    sassOptions: {
-      fiber: Fiber,
-    },
```
Comment (10xLaCroixDrinker) [inline packages/one-app-bundler/webpack/loaders/common.js:72]: This is the only functional change in this PR

File: packages/one-app-runner/__tests__/src/startApp.spec.js [hunk 3]
```diff
@@ -41,33 +41,29 @@ describe('startApp', () => {
...
-  });
-
-  it('pulls one app docker image and starts one app', () => {
-    const mockSpawn = require('mock-spawn')();
+  it('pulls one app docker image and starts one app', async () => {
+    expect.assertions(1);
+    const mockSpawn = makeMockSpawn();
     childProcess.spawn.mockImplementationOnce(mockSpawn);
-    startApp({ moduleMapUrl: 'https://example.com/module-map.json', rootModuleName: 'frank-lloyd-root', appDockerImage: 'one-app:5.0.0' });
+    await startApp({ moduleMapUrl: 'https://example.com/module-map.json', rootModuleName: 'frank-lloyd-root', appDockerImage: 'one-app:5.0.0' });
```
Comment (10xLaCroixDrinker) [inline packages/one-app-runner/__tests__/src/startApp.spec.js:50]: These changes were required to resolve run away promises that threw error in node 16

File: packages/one-app-bundler/package.json [hunk 4]
```diff
@@ -38,7 +38,7 @@
   ],
   "license": "Apache-2.0",
   "peerDependencies": {
-    "babel-eslint": "7.x || 8.x"
+    "babel-eslint": "7.x || 8.x || 10.x"
```
Comment (10xLaCroixDrinker) [inline packages/one-app-bundler/package.json:41]: This is to match the version used by one-app, as mismatched peerDeps cause install failures in node 16

File: packages/holocron-dev-server/__tests__/src/utils/virtual-file-system.spec.js [hunk 5]
```diff
@@ -15,5 +15,5 @@
 import * as vfs from '../../../src/utils/virtual-file-system';
 
 test('exports virtual file system interfaces', () => {
-  expect(Object.keys(vfs)).toMatchSnapshot();
+  expect(Object.keys(vfs).sort()).toMatchSnapshot();
```
Comment (10xLaCroixDrinker) [inline packages/holocron-dev-server/__tests__/src/utils/virtual-file-system.spec.js:18]: Node 12 and 16 put these in different orders if unsorted

File: packages/one-app-runner/__tests__/src/startApp.spec.js [hunk 6]
```diff
@@ -77,166 +73,177 @@ describe('startApp', () => {
...
-    const mockStderr = 'hello erroneous world (stderr)';
-    mockSpawn.setDefault(mockSpawn.simple(0, mockStdout, mockStderr));
+  it('outputs all logs from docker pull and docker run to a file if output file arg is given', async () => {
+    expect.assertions(5);
+    const mockProcess = { on: jest.fn(), stdout: { pipe: jest.fn() }, stderr: { pipe: jest.fn() } };
+    const mockSpawn = jest.fn(() => mockProcess);
     childProcess.spawn.mockImplementationOnce(mockSpawn);
-
-    startApp({
+    createWriteStreamSpy.mockReturnValueOnce('stream');
```
Comment (10xLaCroixDrinker) [inline packages/one-app-runner/__tests__/src/startApp.spec.js:207]: This gets rid of the need for that file that was previously being deleted in `afterAll`

**Review by code-forger — APPROVED** [review]

**Review by Matthew-Mallimo — APPROVED** [review]

**Pull request merged**